### PR TITLE
WIP: Etcm 739

### PR DESCRIPTION
**Pull request by AnastasiiaL**
# Description

BlockFetcher refactored to a typed actor and state partially moved to the behaviours

### CODE REVIEW

File: src/main/scala/io/iohk/ethereum/blockchain/sync/regular/BlockImporter.scala [hunk 1]
```diff
@@ -410,11 +464,37 @@ object BlockImporter {
     def branchResolved(): ImporterState = copy(resolvingBranchFrom = None)
 
     def isResolvingBranch: Boolean = resolvingBranchFrom.isDefined
+
+    def withLastImportedBlock(lastImportedBlock: BigInt): ImporterState = copy(lastImportedBlock = lastImportedBlock)
+
+    def fitsOnTop(blockNumber: BigInt): Boolean = {
+      println("1 " + blockNumber)
+      println("2 " + lastImportedBlock)
```
Comment (leo-bogastry) [inline src/main/scala/io/iohk/ethereum/blockchain/sync/regular/BlockImporter.scala:472]: I guess these println are left behind

File: src/test/scala/io/iohk/ethereum/blockchain/sync/regular/BlockFetcherSpec.scala [hunk 2]
```diff
@@ -23,18 +23,28 @@ import io.iohk.ethereum.security.SecureRandomBuilder
...
+import akka.actor.typed.scaladsl.adapter._
 
+import scala.concurrent._
+import ExecutionContext.Implicits.global
 import java.net.InetSocketAddress
+
+import akka.actor.testkit.typed.FishingOutcome.Complete
+import akka.actor.testkit.typed.scaladsl.ScalaTestWithActorTestKit
+import akka.actor.typed.ActorRef
+import io.iohk.ethereum.blockchain.sync.regular.RegularSync.ProgressProtocol
```
Comment (leo-bogastry) [inline src/test/scala/io/iohk/ethereum/blockchain/sync/regular/BlockFetcherSpec.scala:35]: Can you please organize the imports?

File: src/test/scala/io/iohk/ethereum/blockchain/sync/regular/BlockFetcherSpec.scala [hunk 3]
```diff
@@ -127,8 +137,11 @@ class BlockFetcherSpec
 
       handleFirstBlockBatchHeaders()
 
+//      peersClient.expectMsgPF(hint = "hmmm") { case PeersClient.Request(_, _, _) => () }
+//      peersClient.expectMsgPF(hint = "hmmm") { case PeersClient.Request(_, _, _) => () }
```
Comment (leo-bogastry) [inline src/test/scala/io/iohk/ethereum/blockchain/sync/regular/BlockFetcherSpec.scala:141]: Commented code left behind. Also in line 226

File: src/main/scala/io/iohk/ethereum/blockchain/sync/regular/StateNodeFetcher.scala [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+        log.error("Unexpected error while doing a request", error)
+        responseFallback
+      }
+
+  private def blacklistPeerOnFailedRequest(msg: Any): Unit = msg match {
+    case RequestFailed(peer, reason) => peersClient ! BlacklistPeer(peer.id, reason)
+    case _ => ()
+  }
+
+  private def handleRequestResult(fallback: StateNodeFetcherCommand)(msg: Any): Task[StateNodeFetcherCommand] =
```
Comment (leo-bogastry) [inline src/main/scala/io/iohk/ethereum/blockchain/sync/regular/StateNodeFetcher.scala:107]: I see that the `HeadersFetcher`, `BodiesFetcher` and `StateNodeFetcher` share some logic and in the `makeRequest` and `handleRequestResult` methods, can these be abstracted into a trait?

Comment (AnastasiiaL) [reply src/main/scala/io/iohk/ethereum/blockchain/sync/regular/StateNodeFetcher.scala:107]: sorry I forgot to mark this PR as WIP. this was just trying to make it work. But it didn't, we won't proceed with these changes

**Pull request closed without merging**